webpack/components/AnsibleRoles/components/AnsibleRolesImportButton.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { Dropdown, DropdownToggle, DropdownItem } from '@patternfly/react-core';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { useAPI } from 'foremanReact/common/hooks/API/APIHooks';
+import { importUrl } from '../AnsibleRolesPageHelpers';
+import { showToast } from '../../../toastHelper';
+import {
+  formatError,
+  joinErrors,
+} from '../../AnsibleHostDetail/components/AnsibleVariableOverrides/EditableActionHelper';
+
+export const AnsibleRolesImportButton = () => {
+  let dropdownItems;
+
+  const {
+    response: { results },
+    status,
+  } = useAPI('get', '/api/v2/smart_proxies', {
+    params: { feature: 'Ansible' },
+  });
+
+  const [isOpen, setIsOpen] = React.useState(false);
+  const onToggle = isToggleOpen => {
+    setIsOpen(isToggleOpen);
+  };
+  const onFocus = () => {
+    const element = document.getElementById('toggle-primary');
+    element.focus();
+  };
+  const onSelect = () => {
+    setIsOpen(false);
+    onFocus();
+  };
+
+  if (status === 'PENDING') {
+    dropdownItems = [
+      <DropdownItem isDisabled key="loading_item">
+        {__('Loading...')}
+      </DropdownItem>,
+    ];
+  } else if (status === 'ERROR') {
+    showToast({
+      type: 'error',
+      message: __('There was an error requesting Smart Proxies'),
+    });
+  } else if (status === 'RESOLVED') {
+    dropdownItems = results
+      // .filter(result =>
+      //  result.features.some(feature => feature.name === 'Ansible')
+      // )
```

Comment:
This seem to filter things correctly for me

Reply:
It did, but as Ewoud suggested, this filtering can be done server-side.